build/gen_stub.php

```diff
@@ -602,7 +623,60 @@ public function getFunctionEntry(): string {
         }
     }
 
-    private function getFlagsAsString(): string
+    public function discardInfoForOldPhpVersions(): void {
+        $this->return->type = null;
+        foreach ($this->args as $arg) {
+            $arg->type = null;
+            $arg->defaultValue = null;
+        }
+    }
+}
+
+class PropertyInfo extends MemberInfo
+{
+    /** @var string */
+    public $name;
+    /** @var bool */
+    public $isKnownName;
+    /** @var bool */
+    public $isDeprecated;
+    /** @var Type|null */
+    public $type;
+    /** @var string|null */
+    public $value;
+
+    public function __construct(string $name, int $flags, bool $isKnownName, bool $isDeprecated, ?Type $type, ?string $value)
+    {
+        parent::__construct($flags);
+        $this->name = $name;
+        $this->isKnownName = $isKnownName;
+        $this->isDeprecated = $isDeprecated;
+        $this->type = $type;
+        $this->value = $value;
+    }
+
+    public function discardInfoForOldPhpVersions(): void {
+        $this->type = null;
+    }
+
+    public function getDeclaration(): string {
+        $code = "";
+        $code .= "    zend_declare_property_null(ce, \"$this->name\", sizeof(\"$this->name\") - 1, " . $this->getFlagsAsString() . ")\n";
```

Comment:
This part is WIP. I'd like to first get a green flag if this is the right way to follow. :)